File: 2021/08/06/main.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        r = []
        q = [root]
        
        while q:
            r.append([n.val for n in q])
            q = [c for n in q for c in n.children]

        return r

        

File: 2021/08/06/main.go
package main

/**
 * Definition for a Node.
 * type Node struct {
 *     Val int
 *     Children []*Node
 * }
 */

func levelOrder(root *Node) [][]int {
	d := make(map[int]int)
	level = 0
	def traverse(root, level):
		if not root:
			return
		try:
			d[level].append(root.val)
		except:
			d[level] = [root.val]            
		level += 1
		for child in root.children:
			traverse(child, level)
	traverse(root, level)
	return list(d.values())

	return r
}
